integration-tests/src/test/java/com/datastax/oss/driver/api/core/metadata/SchemaAgreementIT.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2017-2017 DataStax Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.metadata;
+
+import com.datastax.oss.driver.api.core.Cluster;
+import com.datastax.oss.driver.api.core.cql.ResultSet;
+import com.datastax.oss.driver.api.core.session.CqlSession;
+import com.datastax.oss.driver.api.testinfra.ccm.CustomCcmRule;
+import com.datastax.oss.driver.api.testinfra.cluster.ClusterRule;
+import com.datastax.oss.driver.api.testinfra.cluster.ClusterUtils;
+import com.datastax.oss.driver.categories.LongTests;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestName;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Category(LongTests.class)
+public class SchemaAgreementIT {
+
+  private static CustomCcmRule ccm = CustomCcmRule.builder().withNodes(3).build();
+  private static ClusterRule clusterRule =
+      ClusterRule.builder(ccm)
+          .withOptions(
+              "request.timeout = 30s",
+              "load-balancing-policy.class = com.datastax.oss.driver.api.testinfra.loadbalancing.SortingLoadBalancingPolicy")
+          .build();
+
+  @ClassRule public static RuleChain ruleChain = RuleChain.outerRule(ccm).around(clusterRule);
+
+  @Rule public TestName name = new TestName();
+
+  @Test
+  public void should_succeed_when_all_nodes_agree() {
+    ResultSet result = createTable();
+
+    assertThat(result.getExecutionInfo().isSchemaInAgreement()).isTrue();
+    assertThat(clusterRule.cluster().checkSchemaAgreement()).isTrue();
+  }
+
+  @Test
+  public void should_fail_on_timeout() {
```

Comment:
How about lowering `schema-agreement.timeout` to make this test a bit faster?

Reply:
good idea, will do